x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/BoostedTreeParams.java

```diff
@@ -150,6 +155,7 @@ public BoostedTreeParams(@Nullable Double lambda,
         softTreeDepthTolerance = in.readOptionalDouble();
         downsampleFactor = in.readOptionalDouble();
         maxOptimizationRoundsPerHyperparameter = in.readOptionalVInt();
+        earlyStoppingEnabled = in.readOptionalBoolean();
```

Comment:
As for write, this needs a version guard, e.g.:

```
if (in.getVersion().onOrAfter(V_8_0_0)) {
    earlyStoppingEnabled = in.readOptionalBoolean();
} else {
    earlyStoppingEnabled = null;
}
```

Reply:
I changed the type to `boolean`